File: Day-01/elixir/icrayix/day_1.exs
elves =
  for elf <- String.split(File.read!("./input.txt"), "\n\n") do
    String.split(elf, "\n", trim: true)
    |> Enum.map(&String.to_integer/1)
    |> Enum.sum()
  end

# part 1
IO.puts(Enum.max(elves))

# part 2
elves
|> Enum.sort(:desc)
|> Enum.take(3)
|> Enum.sum
|> IO.puts()

File: Day-01/rust/Marie/day1.rs
use std::fs;

fn main() {
    let input = fs::read_to_string("inputs/day1.txt").unwrap();
    let mut elves: Vec<i32> = input
        .split("\n\n")
        .map(|text| {
            text.lines()
                .map(|line| line.parse::<i32>().unwrap())
                .sum::<i32>()
        })
        .collect();

    elves.sort();

    let most = elves.last().unwrap();
    println!("{most}");

    let last_three: i32 = elves.iter().rev().take(3).sum();
    println!("{last_three}");
}
